include/strf/detail/facets/encoding.hpp

```diff
@@ -630,18 +630,25 @@ namespace detail {
 
 constexpr std::size_t global_mini_buffer32_size = 16;
 
-inline char32_t* global_mini_buffer32()
+inline STRF_HD char32_t* global_mini_buffer32()
 {
+#ifdef __CUDA_ARCH__
+    // TODO: Try thinking of something smarter than this code...
+
+    asm("trap;"); // there is no equivalent of "static thread_local" in GPU device-side code
+    return nullptr;
+#else
     thread_local static char32_t buff[global_mini_buffer32_size];
     return buff;
+#endif
```

Comment:
`global_min_buffer32()` is not really necessary. For `__CUDA_ARCH__` you can disable this code, and make `buffered_encoder` and `buffered_size_calculator` ( the only places where it is used ) to use a local buffer instead, analogous to how is done in `basic_streambuf_writer` ( `detail/output_types/std_streambuf.hpp` )